ddtrace/profiling/exporter/http.py

```diff
@@ -138,16 +109,25 @@ def _get_tags(self, service):
         tags.update({k: six.ensure_binary(v) for k, v in user_tags.items()})
         return tags
 
+    @staticmethod
+    def _retry_on_http_5xx(exception):
+        if isinstance(exception, error.HTTPError):
+            return 500 <= exception.code < 600
+        return True
+
     def export(self, events, start_time_ns, end_time_ns):
         """Export events to an HTTP endpoint.
 
         :param events: The event dictionary from a `ddtrace.profiling.recorder.Recorder`.
         :param start_time_ns: The start time of recording.
         :param end_time_ns: The end time of recording.
         """
-        common_headers = {
-            "DD-API-KEY": self.api_key.encode(),
-        }
+        if self.api_key:
+            headers = {
+                "DD-API-KEY": self.api_key.encode(),
+            }
+        else:
+            headers = {}
```

Comment:
There's another part to this that corresponds to adding the container id as a header when available (https://github.com/DataDog/dd-trace-go/blob/afab3a8fcca25d2a37bf9de8870cd8295c840980/profiler/upload.go#L97-L99). 

Would this be something that would fit in this PR as well or better to add it in a new one?

Reply:
I'd do that as a different PR — where does the value of that header come from?

Reply:
Same as in the tracer: https://github.com/DataDog/dd-trace-py/blob/ed831e471961b073d7ae78a80f3af507662a0fa3/ddtrace/api.py#L163